common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Clients/BehalfOfTokenProvider.cs

```diff
@@ -83,13 +83,25 @@ public partial class BehalfOfTokenProvider : ITokenProvider {
             var cache = _store.GetCache($"OID:{user.GetObjectId()}");
             var ctx = CreateAuthenticationContext(_config.InstanceUrl,
                 _config.TenantId, cache);
-            try {
-                var result = await ctx.AcquireTokenAsync(resource,
-                    new ClientCredential(_config.AppId, _config.AppSecret),
-                    new UserAssertion(token, kGrantType, name));
+
+            try {   
+                var result = await ctx.AcquireTokenSilentAsync(resource, _config.AppId);
                 return result.ToTokenResult();
             }
             catch (AdalException ex) {
+                if (ex.ErrorCode == AdalError.FailedToAcquireTokenSilently) {
+                    try {           
```

Comment:
>       try {         [](start = 13, length = 20)

I would avoid imbricating try ... catch blocks. Just return null in case of failure and take the imbricated try...catch outside the catch